warp10/src/main/java/io/warp10/script/functions/FETCH.java

```diff
@@ -1485,4 +1537,167 @@ public static Long getTimestamp(Object timestampRepresentation, String timestamp
 
     return timestamp;
   }
+
+  public static Map<String,Object> parseTokenAttributes(ReadToken token, Map<String,Object> params) throws WarpScriptException {
+    Map<String,Object> attributes = new HashMap<String,Object>();
+
+    Map<String,String> tokenattr = token.getAttributes();
+
+    if (null == tokenattr) {
+      return attributes;
+    }
+
+    long now = TimeSource.getTime();
+
+    Instant ref = new Instant(now / Constants.TIME_UNITS_PER_MS);
+
+    String value = tokenattr.get(Constants.TOKEN_ATTR_END);
+
+    Long end = null;
+
+    if (null != value) {
+      value = value.trim();
+
+      if (value.startsWith("P")) {
+        long offset = DURATION.parseDuration(ref, value, true, true);
+        end = now + offset;
```

Comment:
`DURATION.parseDuration(ref, value, true, true);` will return a positive duration for `DURATION.parseDuration(new Instant(), "PT1S", true, true);`, with or without https://github.com/hbs/warp10-platform/pull/10.

It seems to me all `+ offset` should be `- offset`.